src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -470,8 +470,18 @@ protected String remoteCheckpointDocID() {
 
     public Set<String> getPendingDocumentIDs() {
         synchronized (lockPendingDocIDs) {
-            if (isPull() || (isRunning() && pendingDocIDs != null))
-                return pendingDocIDs;
+
+            /** 
+             * Bug causing code                
+             * Expected Functionality:
+             *      While offline, I expect the return value of this method to reflect
+             *      the number of pendingDocIDs inline with database mutations 
+             * Actual Behavior:
+             *      The commented out lines below, output a cached value for pendingDocIDs 
+             *
+             * if (isPull() || (isRunning() && pendingDocIDs != null))
+             *   return pendingDocIDs;
+            */
```

Comment:
Hi, I can leave in the null check for pull replicator. 

However, once we've determined that we're dealing with a push, we should proceed without returning the cached value of pendingDocIDs (since that will not reflect new updates to lite)

Do you agree ?